### PR TITLE
[ios][image] Support loading assets in native directory

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Closes ENG-11393
Add support for loading images that were added directly to the native project. This was already working on android.

# How
Added some test assets to bare-expo and a new screen in NCL.
Added a new function that checks if the image exists locally and renders it instead of attempting to load from a url.
Noticed an issue in the `isBlurhashString` function where a `?` was used to match the first capture group meaning something like this `image1` would match. Changed it to a `+`. 

# Test Plan
Tested on both platforms in bare-expo. Both work as expected.

| iOS  | Android  |
|---|---|
|  Simulator Screenshot - iPhone 15 Pro - 2024-02-23 at 11 04 15 | Screenshot_1708691009 |

### CODE REVIEW

File: packages/expo-image/src/utils/resolveSources.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { resolveBlurhashString, resolveThumbhashString } from './resolveHashStri
 import { ImageNativeProps, ImageProps, ImageSource } from '../Image.types';
 
 export function isBlurhashString(str: string): boolean {
-  return /^(blurhash:\/)?[\w#$%*+,\-.:;=?@[\]^_{}|~]+(\/[\d.]+)*$/.test(str);
+  return /^(blurhash:\/)+[\w#$%*+,\-.:;=?@[\]^_{}|~]+(\/[\d.]+)*$/.test(str);
```
Comment (lukmccall) [inline packages/expo-image/src/utils/resolveSources.tsx:6]: I believe we should include that in the changelog too, as it appears to be a separate issue, correct?

**Review by lukmccall — APPROVED** [review]

File: packages/expo-image/src/utils/resolveSources.tsx (hunk 1 above)
Comment (alanjhughes) [reply packages/expo-image/src/utils/resolveSources.tsx:6]: Yes, your right, I'll add it 👍

Comment (tsapeta) [reply packages/expo-image/src/utils/resolveSources.tsx:6]: This seems to break detecting if it's a blurhash when it doesn't start with `blurhash:/`. I think `?` was correct here.

Comment (alanjhughes) [reply packages/expo-image/src/utils/resolveSources.tsx:6]: Maybe I didn't understand it was supposed to do then. We want it to take any string, and append `blur hash:/` to the start whether it starts with it or not? That breaks loading a local asset if that's the case. If I give a source of `image1`, I expect it to load a local asset named `image1`. Without this change it gets turned into `blurhash:/image1`.

Comment (alanjhughes) [reply packages/expo-image/src/utils/resolveSources.tsx:6]: So we want to take any string and add blurhash:/? That regex will match everything at the moment.

**Pull request merged**